scripts/coverage.sh

```diff
@@ -32,12 +33,21 @@ generate_cover_data() {
 
     for pkg in "$@"; do
         f="$workdir/$(echo $pkg | tr / -).cover"
-        go test -v -covermode="$mode" -coverprofile="$f" "$pkg"
+        # go test -v -covermode="$mode" -coverprofile="$f" "$pkg"
+        # not using -v verbose flag, too much info to parse easily; output to file
+        go test -covermode="$mode" -coverprofile="$f" "$pkg" >> "$report"
     done
 
     echo "mode: $mode" >"$profile"
     grep -h -v "^mode:" "$workdir"/*.cover | $dir/exclude_ignore.py | egrep -v "$excludes" >>"$profile"
 }
 
-generate_cover_data $(go list ./... | grep -v /vendor/ | grep -v integration )
+generate_cover_data $(go list ./pkg/* | grep -v /vendor/ | grep -v integration )
 go tool cover -html="$profile" -o="$html"
+
+# parse coverage data to a json-friendly format
+report_data=$($dir/parse_coverage_report.py)
+
+# POST report to dashboard API
+# substitute $report_data to curl's input, which need 8 qoutes in total
+curl -X POST "http://35.197.82.212/post-coverage" -d ''"$report_data"''
```

Comment:
that is scary... no authentication? this is public endpoint, everybody can access it. Also this will be executed on every PR, on every update. IMHO this should be updated only when something is merged to master. Should we add new script and CI job?

Reply:
I would also prefer a separate CI job for coverage but if it's updating the dashboard it must be categorized against a PR and not as the coverage of the entire project.  @zimengyang Can you clarify the intent?